src/Orleans/Core/GrainClient.cs

```diff
@@ -381,6 +381,15 @@ public static Streams.IStreamProvider GetStreamProvider(string name)
             return RuntimeClient.Current.CurrentStreamProviderManager.GetProvider(name) as Streams.IStreamProvider;
         }
 
+        public delegate void ConnectionToClusterLostHandler(object sender, EventArgs e);
```

Comment:
If we provide "lost" event, we will most surely need to provide "gained back" event and count. Otherwise, from pure API design perspective this looks half baked. In particular, the rule you should apply is: would it pass MSDN API review?